packages/react-redux-subspace/src/hooks/useSubspace.js

```diff
@@ -6,31 +6,31 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import { useContext, useMemo } from "react"
+import { useRef, useMemo, useContext } from "react"
 import { ReactReduxContext } from "react-redux"
 import { subspace } from "redux-subspace"
-import isObject from '../utils/isObject'
+import isObject from "../utils/isObject"
+import isFunction from "../utils/isFunction"
 
-export const useSubspaceAdvanced = (
-  mapState,
-  namespace,
-  subspaceDecorator,
-  { context = ReactReduxContext } = {}
-) => {
-  const subspacer = useMemo(
-    () => subspaceDecorator || subspace(mapState, namespace),
-    [mapState, namespace]
-  )
-  const { store } = useContext(context)
-  return useMemo(() => subspacer(store), [subspacer, store])
-}
-
-const useSubspace = (mapState, namespace, options) => {
+export const useSubspace = (mapState, namespace, options) => {
   if (options === undefined && isObject(namespace)) {
     return useSubspace(mapState, undefined, namespace)
   }
 
-  return useSubspaceAdvanced(mapState, namespace, undefined, options)
+  const { context = ReactReduxContext } = options || {}
+
+  const mapStateFunc = useRef(mapState)
+  mapStateFunc.current = mapState
+
+  const subspacer = isFunction(mapState)
+    ? useMemo(
+        () => subspace((...args) => mapStateFunc.current(...args), namespace),
+        [mapStateFunc, namespace]
```

Comment:
`mapStateFunc` is never going to change right? are you better off using something more obviously constant to pad out the dependencies array like `true` or (defined elsewhere) `REACT_NEEDS_DEPENDENCIES_ARRAY_TO_NOT_CHANGE_LENGTH`?

(or have a missed the point?)

Reply:
You could do something like that. React doesn't care what the prop is, just that it doesn't change length.

I think I prefer the current solution anyway. While the result of `useRef` is static, our code has no guarantees of that, and the implementation of different renderers might be different. This way, if it did change, we would be covered.

Reply:
thats fair.